### DISCUSSION TITLE
Restructure plugins for sample data

### DISCUSSION THREAD

**Discussion by chmst** [joomla/joomla-cms#47139] [2. Cold Features]
### Is your feature request related to a problem? Please describe.
Sample data plugins were made in J3 but now we have much changes in views / database / features.

Blog sample data have been updated, but this is not a valid 
Now in J4 these plugins (blog sample data, multilanguage, testing) are not up to date and very hard to maintain.
Hard to maintain 

- manily because data and control structure are not separated properly
- Texts are not all matching features and naming in J4
- Coding standads have changed
- and others

### Describe the solution you'd like
A new concept is needed für plugins 
Data must be checked and updated, for **all views** of the whole system
Restructuring should also contain the installation process - the menuItem "home", which then is in relation to mutilanguage data.

A good example of how plugins could be made is here https://github.com/joomla/joomla-cms/discussions/32744#discussioncomment-533950

**Comment by ghost** [comment]
> Now in J4 these plugins (blog sample data, multilanguage, testing)

@chmst can i please ask you where i can get this `testing sample data`? 
I installed J4 using launch.joomla.org having blog sample data, multilanguage but no testing, also don't get `testing` by `Extensions: Discover`.

**Comment by infograf768** [comment]
@Bakual

**Comment by chmst** [comment]
@sandramay0905 if you clone the 4.0 dev branch of Joomla-cms, you get the plugin for testing sample data.

**Comment by ghost** [comment]
> if you clone the 4.0 dev branch 

thanks for the answer, no idea what it mean. I was only looking if i haven't found a plugin.

**Comment by Bakual** [comment]
Texts should actually be simple to adjust. Those are regular language strings.
Coding standards should also be simple.

The control/data thing is actually hard to do in a plugin. Back then I thought about defining the data in an XML file and then read by the plugin. But since some parts have to be defined dynamically (eg the category id), static files don't work well.

As for copying files like suggested in the other thread. Imho it's a bit pointles, but could of course be done easily.
